### ISSUE TITLE
[Update] Cannot translate attribute `heading.*` as it's not one of the translatable attributes

### ISSUE THREAD

**Issue by voidgraphics** [optimistdigital/nova-translatable#8]
Hi, love this package, it's great!

I am having an error when updating a resource that has a translated field within a Flexible Content.

When the flexible is empty and I initially update the resource to add new data, it works great. But when trying to update a resource that already has data in the flexible's translated fields, I get the following error:

```
Cannot translate attribute `heading.*` as it's not one of the translatable attributes: `heading` 
```

It seems to be related to the validator trick that adds the `.*` to the attribute name.

Here's a summary of my code:

```php
public function fields()
{
    return [
        Flexible::make('Cards')->addLayout(CardLayout::class)
    ];
}
```

```php
class CardLayout extends Layout
{
    use HasTranslations;

    public $translatable = [
        'heading'
    ];

    protected $name = 'cardlayout';
    protected $title = 'Card';

    public function fields()
    {
        return [
            Text::make('Heading', 'heading')->translatable(),
        ];
    }

}
```

You mentioned in #7 that you managed to get translatable fields to work with flexible content fields, did you run into this problem? How did you do it?

Thanks for your time!

**Comment by voidgraphics** [comment]
After further investigation, this only happens on nova-page templates, and only if the template uses the `HasTranslations` trait for other translated fields that are not part of the flexible content. My above example would actually work on a regular nova resource.

Closing this now as this is a compatibility bug between two packages and not a bug with your code. My apologies.

**Comment by voidgraphics** [comment]
Ah, this is embarrassing. Further testing showed that it DOES happen on regular nova resources, the confusion came from the fact that the error only occurs when the flexible field has at least 2 layout instances.

So, this works:
![Capture d’écran 2020-02-05 à 16 43 42](https://user-images.githubusercontent.com/9298484/73857180-d4921a00-4836-11ea-97ee-f850fa4dcce9.png)

But this does not:
![Capture d’écran 2020-02-05 à 16 44 10](https://user-images.githubusercontent.com/9298484/73857217-e8d61700-4836-11ea-82ac-7d9328eedc89.png)

**Comment by Tarpsvo** [comment]
Hey! Sorry for the late reply, I was on vacation.

Did you use version 1.3.0? It should've fixed the issue of attributes having the `.*` at the end while saving.

I'll test it out in a second as well and see what's up.

**Comment by voidgraphics** [comment]
No worries! Yes, I am using `1.3.0`.

**Comment by Tarpsvo** [comment]
Hey!

I managed to reproduce the issue and pretty sure I got it fixed. Try out version 1.3.1.

**Comment by Tarpsvo** [comment]
Seems to work as expected now inside Flexible layouts. Let me know if anything else arises, thanks and good luck!

**Comment by voidgraphics** [comment]
Thanks for the fix, I'll try it later and let you know if I still have problems with it! Cheers

**Comment by voidgraphics** [comment]
Hi @Tarpsvo! I am encountering this again.

This time the problem arises regardless of the number of layouts present. But only if I have 3 nested flexible fields. Roughly, something like:


```php
Flexible::make('First level')
    ->addLayout('First level layout', 'one', [
        Flexible::make('Second level')
            ->addLayout('Second level layout', 'two', [
                Flexible::make('Third level')
                    ->addLayout('Third level layout', 'three', [
                        Text::make('Title')->translatable()
                    ])
            ])
    ])
```

Something like this, but with proper Layout classes instead of course to be able to specify the `HasTranslations` trait etc.

I can maybe submit a PR for a fix, but since you've already dealt with this before, maybe you can point me in the right direction?

Thanks!

**Comment by Tarpsvo** [comment]
My lord that's a lot nested layouts. Is this something that broke recently and worked previously? Might have to take a look at my recent commits.

**Comment by voidgraphics** [comment]
> My lord that's a lot nested layouts.

Yeah... I don't love it, but that's how I have to structure this one, sadly. I really can't say if it worked before or not, as I try to avoid this much nesting usually. What I can say is that it's the exact same error message as I had when I originally opened this issue!

**Comment by voidgraphics** [comment]
Now I'm seeing this in a non-nested flexible field as well.. Very strange. It works fine for one resource and not for the other. I have to do more digging.

**Comment by voidgraphics** [comment]
I have made some progress!

In `TranslatableFieldMixin.php`, there is [a trick](https://github.com/optimistdigital/nova-translatable/blob/master/src/TranslatableFieldMixin.php#L125) to play nice with the validator : we append `.*` to the attribute name and saves it in `$this->attribute`. Normally, this works fine and does the trick.

However it seems that when it loops over the flexible layouts,  the instance is kept during the loop - meaning that when it gets to the second layout, the trick has already been applied, but this code does not know that and applies it again. So we end up with a `.*.*` after the attribute name, which is bad. 

I have verified this with the `spl_object_hash` function which gives a unique hash per instance, and the same hash comes out twice just before it errors out, during the loop over the flexible layouts.

So I tried a workaround: keep track of the `spl_object_hash` of the objects that have already had the trick applied, and prevent it from applying it twice:

```php
if(!request()->tricked) {
    request()->tricked = [];
}

// If it's a CREATE or UPDATE request, we need to trick the validator a bit
if (in_array(request()->method(), ['PUT', 'POST']) && !in_array(spl_object_hash($this), request()->tricked)) {
    $this->attribute = "{$this->attribute}.*";
    request()->tricked []= spl_object_hash($this);
}
```

This is more a proof than a proper solution, in my eyes. It's probably a bit dirty to store those in the request, it was just the easiest way to debug it quickly.
